platform/import-export/src/actions/ImportAsJSONAction.js

```diff
@@ -130,8 +136,23 @@ define(['zepto'], function ($) {
      * @private
      */
     ImportAsJSONAction.prototype.rewriteId = function (oldID, newID, tree) {
-        tree = JSON.stringify(tree).replace(new RegExp(oldID, 'g'), newID);
-        return JSON.parse(tree);
+        let newIdKeyString = this.openmct.objects.makeKeyString(newID);
+        let oldIdKeyString = this.openmct.objects.makeKeyString(oldID);
+        tree = JSON.stringify(tree).replace(new RegExp(oldIdKeyString, 'g'), newIdKeyString);
+
+        return JSON.parse(tree, (key, value) => {
+            if (Object.prototype.hasOwnProperty.call(value, 'key') &&
+                Object.prototype.hasOwnProperty.call(value, 'namespace') &&
+                value.key === oldID.key &&
+                value.namespace === oldID.namespace) {
+                return {
+                    namespace: newID.namespace,
+                    key: newID.key
+                }
```

Comment:
Can't newID be returned here instead? This object seems to have the same property/value as the object in line 120, no?